src/AppInstallerRepositoryCore/Public/winget/RepositorySearch.h

```diff
@@ -351,31 +334,18 @@ namespace AppInstaller::Repository
         // Gets the installed package information.
         virtual std::shared_ptr<IPackageVersion> GetInstalledVersion() const = 0;
 
-        // Note on pins:
-        // Pins only make sense when there is both an installed and an available version.
-        // Only for the composite source will GetAvailableVersionKeys() include pinned state,
-        // and GetLatestAvailableVersion() consider the pin behavior.
-
         // Gets all available versions of this package.
         // The versions will be returned in sorted, descending order.
         //  Ex. { 4, 3, 2, 1 }
         // The list may contain versions from multiple sources.
-        virtual std::vector<PackageVersionKey> GetAvailableVersionKeys(PinBehavior pinBehavior = PinBehavior::ConsiderPins) const = 0;
+        virtual std::vector<PackageVersionKey> GetAvailableVersionKeys() const = 0;
 
         // Gets a specific version of this package.
-        virtual std::shared_ptr<IPackageVersion> GetLatestAvailableVersion(PinBehavior pinBehavior) const = 0;
+        virtual std::shared_ptr<IPackageVersion> GetLatestAvailableVersion() const = 0;
 
         // Gets a specific version of this package.
         virtual std::shared_ptr<IPackageVersion> GetAvailableVersion(const PackageVersionKey& versionKey) const = 0;
 
-        virtual std::pair<std::shared_ptr<IPackageVersion>, Pinning::PinType> GetAvailableVersionAndPin(const PackageVersionKey& versionKey) const
-        {
-            return { GetAvailableVersion(versionKey), Pinning::PinType::Unknown };
-        }
-
-        // Gets a value indicating whether an available version is newer than the installed version.
-        virtual bool IsUpdateAvailable(PinBehavior pinBehavior) const = 0;

```

Comment:
Correct, it didn't seem right to leave this there to be misused.  Additionally, it can't properly consider multiple installed versions, which is a major factor driving this change.